x/profiles/keeper/keeper_chain_link.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	sdkquery "github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+// StoreChainLink stores the given chain link inside the current context.
+// It assumes that the given chain link has already been validated.
+func (k Keeper) StoreChainLink(ctx sdk.Context, link types.ChainLink, destinationAddress string) error {
+	if _, found := k.GetChainLink(ctx, link.ChainConfig.Name, link.Address); found {
+		return fmt.Errorf("chain link already exists")
+	}
+
+	// Check if address has the profile
+	profile, found, err := k.GetProfile(ctx, destinationAddress)
+	if err != nil {
+		return err
+	}
+	if !found {
+		return fmt.Errorf("address does not have any profile")
+	}
+	// Store chain link to the profile
+	profile.ChainsLinks = append(profile.ChainsLinks, link)
+	if err := k.StoreProfile(ctx, profile); err != nil {
+		return err
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	key := types.ChainsLinksStoreKey(link.ChainConfig.Name, link.Address)
+	store.Set(key, k.cdc.MustMarshalBinaryBare(&link))
+	return nil
+}
+
+// GetChainLink returns the chain link corresponding to the given address and the given chain name inside the current context.
+func (k Keeper) GetChainLink(ctx sdk.Context, chainName string, address string) (link types.ChainLink, found bool) {
+	store := ctx.KVStore((k.storeKey))
+
+	bz := store.Get(types.ChainsLinksStoreKey(chainName, address))
+	if bz == nil {
+		return types.ChainLink{}, false
+	}
+	k.cdc.MustUnmarshalBinaryBare(bz, &link)
+	return link, true
+}
+
+// GetAllChainsLinks returns a list of all the chains links inside the given context.
+func (k Keeper) GetAllChainsLinks(ctx sdk.Context) []types.ChainLink {
+	var links []types.ChainLink
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.ChainsLinksPrefix)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var link types.ChainLink
+		k.cdc.MustUnmarshalBinaryBare(iterator.Value(), &link)
+		links = append(links, link)
+	}
+	return links
+}
+
+// GetChainLinksWithPagination returns a list of the links which is paginated from all the links
+func (k Keeper) GetChainsLinksWithPagination(ctx sdk.Context, page int, limit int) []types.ChainLink {
+	links := k.GetAllChainsLinks(ctx)
+	if page == 0 {
+		page = 1
+	}
+	var pagedLinks []types.ChainLink
+	start, end := client.Paginate(len(links), page, limit, sdkquery.DefaultLimit)
+	if start < 0 || end < 0 {
+		pagedLinks = []types.ChainLink{}
+	} else {
+		pagedLinks = links[start:end]
+	}
+	return pagedLinks
+}
+
+// GetUserChainsLinks returns a list of links by a given address and pagination params
+func (k Keeper) GetUserChainsLinks(ctx sdk.Context, address string, page int, limit int) ([]types.ChainLink, error) {
```

Comment:
I think we can safely remove this query. Since chain links for a user are stored inside their profile, if someone wants to know them they can simply use the `desmos query auth account` command and it would return the entire list. 

It would be a lot more useful to have a query that allows you to get the profile based on a given chain id and profile: 

```golang
type QueryProfileByLink struct {
  ChainID string
  Address string
}
```

(This also applies to the CLI and gRPC queries)

Reply:
@RiccardoM In current chain link store logic and its struct, the chain link does not include the `destAddr` that can retrieve the profile easily.  Should it have a prefix to store the relationship between chain link and profile? Say:
```go
func (k Keeper) StoreLinkToProfile(ctx, profile, link) {
    store := ctx.KVStore(k.storeKey)
    key := types.ChainLinkToProfileStoreKey(llink.Address)
    store.Set(key, []byte(profile.GetAddress().String()))
}
``` 

Or adding new field named `destAddr` to `ChainLink` is better than above one?

Reply:
@dadamu In my head, what I think we should do is the following: 

1. Store the chain link inside the `Profile` object.  
   This makes it super easy to get all the links given a Desmos address associated with a profile. 
2. Store the `Profile` address associating it to the chain link address.  
   This makes it super easy to get the `Profile` object given a chain address.

Summarizing, I think we can store: 

```
Profile -> []ChainLink
ChainLink.Address.String() -> Profile.Address
```

Let me know if there's something not clear

Reply:
@RiccardoM The `Address` of  `ChainLink` only stores source address now. It is like:
```proto
message ChainLink {
  Address address = 1; 
  // ...
}
message Address {
  oneof sum {
    Bech32 bech_32 = 1;
    Base58 base_58 = 2;

}
```
That's why I think It should have a new field `destination` to be related to `Profile`. Say:
```proto
message ChainLink {
  Address address = 1; 
  // ...
  string destination = 4;
  Timestamp creation_time = 5;
}
```

Reply:
@dadamu There's no need to store a destination address inside the `ChainLink` object itself. We can simply avoid storing such object and simply store the following association: 

Inside package `types`
```golang
var (
  ChainLinkPrefix = []byte("chain_link")
)

// ChainLinkKey returns the key used to identify the link to the given chain for the given address
func ChainLinkKey(chainName, chainAddress string) []byte {
  return appen(ChainLinkPrefix, bytes(chainName + chainAddress)
}
```

Then we can store the association like this:
```golang
// Store the link -> address association
store := ctx.KVStore(k.storeKey)
store.Set(types.ChainLinkKey(link.ChainName, link.Address.String(), profile.GetAddress())
```

And when a user asks the profile associated with a specific chain address they can use the following command: 

```
desmos q profiles chain-link [chain-name] [chain-address]
```

Which can be implemented as follows: 

```golang
store := ctx.KVStore(k.storeKey)
bz := store.Get(types.ChainLinkKey(chainName, chainAddress)
profile := k.accountKeeper.GetAccount(sdk.AccAddress(bz)) 
```

Reply:
@RiccardoM With it, I think we should also remove `ChainsLinks` method that query all chains links cause the links detail are stored in `Profile` object.

Reply:
@dadamu Yes, I agree we should remove that as well